### PR TITLE
Reinstate `linea_estimateGas` page with caveats

### PR DESCRIPTION
**Pull request by jlwllmr**
- Re-adds the reference section 
- Reinstates `linea_estimateGas` reference page
- Adds a callout on the page advising that the method will only work with Linea Besu, and is not yet available on mainnet.
- Minor additions to the main gas estimation page to emphasise that `linea_estimateGas` is the best method to use.

### CODE REVIEW

File: docs/developers/guides/gas/gas-fees.mdx [hunk 1]
```diff
@@ -42,18 +42,67 @@ components:
 
 ## Estimating transaction costs
 
-:::warning Method deactivated
+Use the [`linea_estimateGas`](/developers/reference/api/linea-estimategas) API method to estimate the
```
Comment (bgravenorst) [inline docs/developers/guides/gas/gas-fees.mdx:45]: We should probably mention that it's currently only available on Linea Sepolia.

File: docs/developers/guides/gas/gas-fees.mdx [hunk 2]
```diff
@@ -42,18 +42,67 @@ components:
...
-Generates and returns an estimate of how much gas is necessary to allow the transaction to complete.
-- [`eth_gasPrice`](https://docs.infura.io/api/networks/linea/json-rpc-methods/eth_gasprice) — Returns 
-the current gas price in wei.
-- [`eth_feeHistory`](https://docs.infura.io/api/networks/linea/json-rpc-methods/eth_feehistory) — 
-Returns historical gas information.
+<Tabs className="my-tabs">
+  <TabItem value="curl" label="curl" default>
+
+```bash
+curl -X POST --data '{"jsonrpc": "2.0","method": "linea_estimateGas","params": [{"from": "0xFE3B557E8Fb62b89F4916B721be55cEb828dBd73","gasPrice":"0x100000000","gas":"0x21000"}],"id": 53}' https://rpc.linea.build
```
Comment (bgravenorst) [inline docs/developers/guides/gas/gas-fees.mdx:72]: This won't work until it's activated on mainnet, so should we use the sepolia endpoint?

File: docs/developers/reference/api/linea-estimategas.mdx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import TabItem from '@theme/TabItem';
+
+# `linea_estimateGas`
+
+:::info
+
+`linea_estimateGas` is being reactivated on testnet from May 28. Mainnet activation is scheduled
+for July 24, 2024. 
+
+It will only be fully compatible with endpoints using the Besu client — if it is unavailable, try
```
Comment (bgravenorst) [inline docs/developers/reference/api/linea-estimategas.mdx:16]: Will this be available for users running a Besu follower node? (our existing run a node instructiions) If not, we may want to clarify.

File: docs/developers/reference/api/linea-estimategas.mdx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# `linea_estimateGas`
+
+:::info
+
+`linea_estimateGas` is being reactivated on testnet from May 28. Mainnet activation is scheduled
+for July 24, 2024.
```
Comment (bgravenorst) [inline docs/developers/reference/api/linea-estimategas.mdx:14]:
```suggestion
`linea_estimateGas` is currently only available on the Linea Sepolia testnet. Mainnet activation is scheduled
for July 24, 2024. 
```

File: docs/developers/reference/api/linea-estimategas.mdx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+:::tip
+
+You can also call the API using [Infura's supported Linea endpoints](https://docs.infura.io/api/networks/linea/choose-a-network).
+:::
+
+<Tabs>
+  <TabItem value="cURL request">
+
+   ```bash
+  curl https://rpc.linea.build \
```
Comment (bgravenorst) [inline docs/developers/reference/api/linea-estimategas.mdx:77]: Should we use the Sepolia endpoint instead?

**Review by bgravenorst — APPROVED** [review]
Added some comments.

File: docs/developers/reference/api/linea-estimategas.mdx (hunk 3 above)
Comment (jlwllmr) [reply docs/developers/reference/api/linea-estimategas.mdx:16]: Yes, my understanding is that this is the case

**Pull request merged**